tools/sst_dump_tool.cc

```diff
@@ -216,9 +220,33 @@ int SstFileDumper::ShowAllCompressionSizes(
           imoptions, moptions, ikc, &block_based_table_factories, i.first,
           compress_opt, nullptr /* compression_dict */,
           false /* skip_filters */, column_family_name, unknown_level);
-      uint64_t file_size = CalculateCompressedTableSize(tb_opts, block_size);
-      fprintf(stdout, "Compression: %s", i.second);
-      fprintf(stdout, " Size: %" PRIu64 "\n", file_size);
+      uint64_t num_data_blocks = 0;
+      uint64_t file_size = CalculateCompressedTableSize(tb_opts, block_size,
+                                                        &num_data_blocks);
+      fprintf(stdout, "Compression: %-24s", i.second);
+      fprintf(stdout, " Size: %10" PRIu64, file_size);
+      fprintf(stdout, " Blocks: %6" PRIu64, num_data_blocks);
+      const uint64_t compressed_blocks =
+        opts.statistics->getAndResetTickerCount(NUMBER_BLOCK_COMPRESSED);
+      const uint64_t not_compressed_blocks =
+        opts.statistics->getAndResetTickerCount(NUMBER_BLOCK_NOT_COMPRESSED);
+      if( (compressed_blocks + not_compressed_blocks) > num_data_blocks ){
+        num_data_blocks = compressed_blocks + not_compressed_blocks;
```

Comment:
Could you add a comment on when `(compressed_blocks + not_compressed_blocks) > num_data_blocks` is true? Thanks

Reply:
I added a comment, it is when the option enable_index_compression is true.